### PR TITLE
Search options now settable through a dictionary in python

### PR DESCRIPTION
**Pull request by RyanUnderhill**
Added C APIs to do the equivalent by name/values
Enabled min_length & repetition_penalty search options
Renamed 'search_params' to 'params' as there is now a separate 'search' member. To avoid confusion

Example difference (assuming params is already a GeneratorParams)

```
params.set_search_options({"max_length":256, "min_length":20})
```
old:
```
params.max_length=256
(no equivalent for min_length)
```

### CODE REVIEW

File: src/ort_genai_c.h [hunk 1]
```diff
@@ -137,13 +137,8 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaCreateGeneratorParams(const OgaModel* mode
...
 
-/*
- * \brief Sets the maximum length that the generated sequence can have.
- * \param[in] params The generator params to set the maximum length on.
- * \param[in] max_length The maximum length of the generated sequences.
- * \return OgaResult containing the error message if the setting of the maximum length failed.
- */
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetMaxLength(OgaGeneratorParams* generator_params, size_t max_length);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetSearchNumber(OgaGeneratorParams* generator_params, const char* name, double value);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetSearchBool(OgaGeneratorParams* generator_params, const char* name, bool value);
```
Comment (yufenglee) [inline src/ort_genai_c.h:141]: how about 2 overload functions with name OgaGeneratorParamsSetSearchOptions?

Comment (yufenglee) [reply src/ort_genai_c.h:141]: Just like what C# APIs

Comment (yufenglee) [reply src/ort_genai_c.h:141]: Never mind. C doesn't support overload

Comment (RyanUnderhill) [reply src/ort_genai_c.h:141]: I would, but C can't do overloaded functions. No parameter name mangling in C.

File: src/ort_genai_c.cpp [hunk 2]
```diff
@@ -105,10 +105,24 @@ OgaResult* OGA_API_CALL OgaCreateGeneratorParams(const OgaModel* model, OgaGener
 }
 
 OgaResult* OGA_API_CALL OgaGeneratorParamsSetMaxLength(OgaGeneratorParams* params, size_t max_length) {
```
Comment (yufenglee) [inline src/ort_genai_c.cpp:107]: remove this function

Comment (RyanUnderhill) [reply src/ort_genai_c.cpp:107]: It's already removed. You might be looking at before Baiju's change

Comment (RyanUnderhill) [reply src/ort_genai_c.cpp]: Done

**Review by yufenglee — APPROVED** [review]

**Pull request merged**